internal/client/tanzupackagerepository/package_repository.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package tanzupackagerepositoryclient
+
+import (
+	"net/url"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/transport"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	pkgrepository "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/tanzupackagerepository"
+)
+
+const (
+	apiVersionAndGroup                 = "v1alpha1/clusters"
+	apiKind                            = "tanzupackage/repositories"
+	namespaces                         = "namespaces"
+	queryParamKeyManagementClusterName = "fullName.managementClusterName"
+	queryParamKeyProvisionerName       = "fullName.provisionerName"
+)
+
+// New creates a new repository resource service API client.
+func New(transport *transport.Client) ClientService {
+	return &Client{Client: transport}
+}
+
+/*
+Client for repository resource service API.
+*/
+type Client struct {
+	*transport.Client
+}
+
+type ClientService interface {
+	RepositoryResourceServiceCreate(request *pkgrepository.VmwareTanzuManageV1alpha1ClusterNamespaceTanzupackageRepositoryRepositoryRequest) (*pkgrepository.VmwareTanzuManageV1alpha1ClusterNamespaceTanzupackageRepositoryResponse, error)
+
+	RepositoryResourceServiceDelete(fn *pkgrepository.VmwareTanzuManageV1alpha1ClusterNamespaceTanzupackageRepositoryFullName) error
+
+	RepositoryResourceServiceGet(fn *pkgrepository.VmwareTanzuManageV1alpha1ClusterNamespaceTanzupackageRepositoryFullName) (*pkgrepository.VmwareTanzuManageV1alpha1ClusterNamespaceTanzupackageRepositoryGetResponse, error)
+
+	RepositoryResourceServiceUpdate(request *pkgrepository.VmwareTanzuManageV1alpha1ClusterNamespaceTanzupackageRepositoryRepositoryRequest) (*pkgrepository.VmwareTanzuManageV1alpha1ClusterNamespaceTanzupackageRepositoryResponse, error)
+}
+
+/*
+RepositoryResourceServiceCreate creates a repository.
+*/
+func (c *Client) RepositoryResourceServiceCreate(request *pkgrepository.VmwareTanzuManageV1alpha1ClusterNamespaceTanzupackageRepositoryRepositoryRequest) (*pkgrepository.VmwareTanzuManageV1alpha1ClusterNamespaceTanzupackageRepositoryResponse, error) {
+	requestURL := helper.ConstructRequestURL(apiVersionAndGroup, request.Repository.FullName.ClusterName, namespaces, request.Repository.FullName.NamespaceName, apiKind).String()
```

Comment:
Dont we need to validate if the `ClusterName` and `NamespaceName` is not empty in `FullName`?

Reply:
I am taking care of these 2 under the operational handlers functions, so here it's not needed.